fitsio/fitslib.py

```diff
@@ -135,9 +135,37 @@ def read_header(filename, ext=0, extver=None, case_sensitive=False, **keys):
     case_sensitive: bool, optional
         Match extension names with case-sensitivity.  Default is False.
     """
-    item=_make_item(ext,extver=extver)
-    with FITS(filename, case_sensitive=case_sensitive) as fits:
-        return fits[item].read_header()
+
+    dont_create=0
+    _fits =  _fitsio_wrap.FITS(filename, READONLY, dont_create)
+    try:
+        hdunum = ext+1
+    except TypeError:
+        extname=mks(ext).upper()
```

Comment:
Do we need to handle the case_sensitive keyword here?

Reply:
probably, thanks